packages/lesswrong/lib/vulcan-lib/fragments.ts

```diff
@@ -1,13 +1,16 @@
 import type { DocumentNode } from 'graphql';
 import gql from 'graphql-tag';
 import * as _ from 'underscore';
-import SqlFragment from '../sql/SqlFragment';
+
+// This is safe as it uses `import type`
+// eslint-disable-next-line import/no-restricted-paths
+import type SqlFragment from '@/server/sql/SqlFragment';
```

Comment:
Apparently if we switch to `@typescript-eslint/no-restricted-imports` we could [allow type imports](https://typescript-eslint.io/rules/no-restricted-imports/#allowtypeimports) in general. Up to you to add this if you're feeling particularly public spirited

Reply:
I tried to do this but the problem is this is a `no-restricted-paths` rule, not `no-restricted-imports`, and I don't think it supports `allowTypeImports`. I'm going to merge for now but if it turns out I'm wrong and there's some way to do this (which I hope there is) we can just make another PR.